src/org/opensolaris/opengrok/history/HistoryCache.java

```diff
@@ -126,6 +126,12 @@ String getLatestCachedRevision(Repository repository)
     void clear(Repository repository) throws HistoryException;
 
     /**
+     * Clear entry for single file from history cache.
+     * @param file path to the file
+     */
+    void clearFile(String file);
```

Comment:
It mimicks the `clear()` interface. It just logs the error and goes on. The same thing is done for removal of xref files. It can be changed to `boolean` or `int` and just ignore the values in `IndexDatabase#removeFile()` however this seems unnecessary to me.